src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  removeAnimal(animal) {
+    const animalIndex = Animal.alive.indexOf(animal);
+
+    Animal.alive.splice(animalIndex, 1);
+  }
```

Comment:
[CODE PERFOMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method instead of using `indexOf` and `splice` which can lead to unnecessary performance overhead due to nesting loops.

Reply:
> [CODE PERFOMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method instead of using `indexOf` and `splice` which can lead to unnecessary performance overhead due to nesting loops.

Good point